### PR TITLE
Replace jsx with jiffy

### PR DESCRIPTION
**Pull request by aleklisi**
This PR unifies json library usages, by replacing jsx with jiffy.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6119.1](https://travis-ci.org/esl/MongooseIM/jobs/483804895) / Erlang 19.3 / small_tests / 7552b57d801af72741fc8b92baa2260341bc9489
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6119/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6119/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-24_09.36.25/lib.mongooseim.logs/run.2019-01-24_09.36.26/suite.log.html)

**Comment by mongoose-im** [comment]
[6121.1](https://travis-ci.org/esl/MongooseIM/jobs/483857904) / Erlang 19.3 / small_tests / e659835441cf362998633d550e6bfeb256aab898
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6121/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6121/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-24_12.09.00/lib.mongooseim.logs/run.2019-01-24_12.09.01/suite.log.html)

**Comment by mongoose-im** [comment]
[6122.1](https://travis-ci.org/esl/MongooseIM/jobs/483875678) / Erlang 19.3 / small_tests / 6370ce0d6434b6c4313fd601a5be8379fb95dea4
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6122/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6122/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-24_12.58.26/lib.mongooseim.logs/run.2019-01-24_12.58.27/suite.log.html)

**Comment by mongoose-im** [comment]
[6123.1](https://travis-ci.org/esl/MongooseIM/jobs/483913879) / Erlang 19.3 / small_tests / ab11e33de6c33d9123f1153472fb0c752d82526d
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6123/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6123/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-24_14.25.38/lib.mongooseim.logs/run.2019-01-24_14.25.39/suite.log.html)

**Comment by aleklisi** [comment]
I cannot fully remove jsx form repo because it is used by some dependencies. 
Run `rebar3 tree` to see which.
It is used by `erlcloud`.

**Comment by mongoose-im** [comment]
[6124.1](https://travis-ci.org/esl/MongooseIM/jobs/484243572) / Erlang 19.3 / small_tests / c5cf04380f27f013b2a0ee7f34040d7e314c8cfb
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6124/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6124/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-25_07.31.57/lib.mongooseim.logs/run.2019-01-25_07.31.58/suite.log.html)

File: src/pubsub/mod_pubsub_db_rdbms.erl [hunk 1]
```diff
@@ -266,15 +266,17 @@ set_node(#pubsub_node{nodeid = {Key, Name}, id = undefined, type = Type,
                       owners = Owners, options = Opts, parents = Parents}) ->
     OwnersJid = [jid:to_binary(Owner) || Owner <- Owners],
     {ok, Nidx} = mod_pubsub_db_rdbms_sql:insert_pubsub_node(encode_key(Key), Name, Type,
-                                                     jsx:encode(OwnersJid),
-                                                     jsx:encode(Opts)),
+                                                     jiffy:encode({OwnersJid}),
```
Comment (michalwski) [inline src/pubsub/mod_pubsub_db_rdbms.erl]: Why the surrounding `{}`. This produces different output the before. 
Hint: here, and in other such places we want to produce a list of strings.

File: src/pubsub/mod_pubsub_db_rdbms.erl [hunk 2]
```diff
@@ -266,15 +266,17 @@ set_node(#pubsub_node{nodeid = {Key, Name}, id = undefined, type = Type,
...
+                                                     jiffy:encode({Opts})),
```
Comment (michalwski) [inline src/pubsub/mod_pubsub_db_rdbms.erl]: We probably don't need surrounding `{}` for `Opts` either, do we?

**Comment by mongoose-im** [comment]
[6126.1](https://travis-ci.org/esl/MongooseIM/jobs/484269290) / Erlang 19.3 / small_tests / d8500e57a7a634359cb9a849917825c988bdd640
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6126/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6126/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-25_09.12.43/lib.mongooseim.logs/run.2019-01-25_09.12.44/suite.log.html)

File: src/pubsub/mod_pubsub_db_rdbms.erl (hunk 2 above)
Comment (aleklisi) [reply src/pubsub/mod_pubsub_db_rdbms.erl]: I think we do. See:
https://github.com/talentdeficit/jsx#to-convert-an-erlang-term-into-a-utf8-binary-containing-a-json-string
```erlang
1> jsx:encode([{<<"library">>,<<"jsx">>},{<<"awesome">>,true}]).
<<"{\"library\": \"jsx\", \"awesome\": true}">>
```
https://github.com/davisp/jiffy#usage
```erlang
2> Doc = {[{foo, [<<"bing">>, 2.3, true]}]}.
{[{foo,[<<"bing">>,2.3,true]}]}
3> jiffy:encode(Doc).
<<"{\"foo\":[\"bing\",2.3,true]}">>
```

**Comment by mongoose-im** [comment]
[6127.1](https://travis-ci.org/esl/MongooseIM/jobs/484297509) / Erlang 19.3 / small_tests / 10d06bafba08f7174a6fe5b091b8277269749370
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2199/6127/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2199/6127/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-25_10.29.51/lib.mongooseim.logs/run.2019-01-25_10.29.52/suite.log.html)

**Pull request merged**